v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/transforms/PubSubMessageToJsonDocument.java

```diff
@@ -0,0 +1,82 @@
+package com.google.cloud.teleport.v2.elasticsearch.transforms;
+
+import com.google.auto.value.AutoValue;
+import com.google.cloud.teleport.v2.transforms.JavascriptTextTransformer;
+import com.google.cloud.teleport.v2.values.FailsafeElement;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.apache.beam.sdk.values.TupleTagList;
+
+/**
+ * The {@link PubSubMessageToJsonDocument} class is a {@link PTransform} which transforms incoming
+ * {@link PubsubMessage} objects into JSON objects for insertion into Elasticsearch while applying
+ * an optional UDF to the input. The executions of the UDF and transformation to Json objects is
+ * done in a fail-safe way by wrapping the element with it's original payload inside the {@link
+ * FailsafeElement} class. The {@link PubSubMessageToJsonDocument} transform will output a {@link
+ * PCollectionTuple} which contains all output and dead-letter {@link PCollection}.
+ *
+ * <p>The {@link PCollectionTuple} output will contain the following {@link PCollection}:
+ *
+ * <ul>
+ *   <li>{@link PubSubToElasticsearch#TRANSFORM_OUT} - Contains all records successfully converted
+ *       to JSON objects.
+ *   <li>{@link PubSubToElasticsearch#TRANSFORM_DEADLETTER_OUT} - Contains all {@link
+ *       FailsafeElement} records which couldn't be converted to table rows.
+ * </ul>
+ */
+@AutoValue
+abstract static class PubSubMessageToJsonDocument
```

Comment:
Since this transformation is specific to PubSub To ElasticSearch can be moved to respective module